### PR TITLE
Feature/pplus 1141 / divided interface

### PR DESCRIPTION
**Pull request by VladislavStrelchenko**

### CODE REVIEW

**Comment by kalaputsik** [comment]
I suppose the structure below might make our source more readable

```
GlueRelationship
├── AbstractModifier.php
├── Unwire
│   ├── UnwireModifierInterface.php
│   ├── UnwireModifier.php
├── Wire
│   ├── WireModifierInterface.php
│   ├── WireModifier.php
CommonClass
├── CommonClassModifierInterface.php
├── CommonClassModifier.php
ClassConstant
├── ModifierInterface.php
├── Modifier.php
├── Wire
│   ├── WireModifierInterface.php
│   ├── WireModifier.php
├── Unwire
│   ├── UnwireModifierInterface.php
│   ├── UnwireModifier.php
ClassInstance
├── Wire
│   ├── WireModifierInterface.php
│   ├── WireModifier.php
├── Unwire
│   ├── UnwireModifierInterface.php
│   ├── UnwireModifier.php
ClassInstanceModifierStrategy
├── Applicable
│   ├── ApplicableReturnArrayModifierStrategy.php
│   ├── ApplicableReturnChainedCollectionModifierStrategy.php
│   ├── ApplicableReturnClassModifierStrategy.php
│   ├── ApplicableReturnCollectionModifierStrategy.php
│   ├── ApplicableReturnExtendContainerModifierStrategy.php
│   ├── ApplicableInterface.php
├── Unwire
│   ├── UnwireModifierStrategyInterfaceCheck.php
│   ├── UnwireReturnArrayModifierStrategy.php
│   ├── UnwireReturnChainedCollectionModifierStrategy.php
│   ├── UnwireReturnClassModifierStrategy.php
│   ├── UnwireReturnCollectionModifierStrategy.php
│   ├── UnwireReturnExtendContainerModifierStrategy.php
├── Wire
│   ├── WireModifierStrategyInterfaceCheck.php
│   ├── WireReturnArrayModifierStrategy.php
│   ├── WireReturnChainedCollectionModifierStrategy.php
│   ├── WireReturnClassModifierStrategy.php
│   ├── WireReturnCollectionModifierStrategy.php
│   ├── WireReturnExtendContainerModifierStrategy.php
```

**Comment by kalaputsik** [comment]
@VladislavStrelchenko 
could you check the methods below in the IntegratorFactory and rename them by template **create{ClassName}** 

```
createCopyFileManifestStrategy
createNavigationManifestStrategy
createMethodBodyCreator
createNodeTreeCreator
createClassConstantModifier
createClassBuilderFactory
createWireClassInstanceReturnArrayModifierStrategy
createWireClassInstanceReturnClassModifierStrategy
createUnwireClassInstanceReturnArrayModifierStrategy
createUnwireClassInstanceReturnCollectionModifierStrategy
createUnwireClassInstanceReturnExtendContainerModifierStrategy
createUnwireClassInstanceReturnChainedCollectionModifierStrategy
createUnwireClassInstanceReturnClassModifierStrategy
createCheckApplicableClassInstanceReturnArrayModifierStrategy
createCheckApplicableClassInstanceReturnCollectionModifierStrategy
createCheckApplicableClassInstanceReturnExtendContainerModifierStrategy
createCheckApplicableClassInstanceReturnChainedCollectionModifierStrategy
createCheckApplicableClassInstanceReturnClassModifierStrategy
createArgumentBuider
```

File: src/IntegratorFactory.php [hunk 1]
```diff
@@ -672,57 +719,176 @@ public function getManifestExecutorStrategies(): array
     }
 
     /**
-     * @return array<\SprykerSdk\Integrator\Builder\ClassModifier\ClassInstanceModifierStrategy\ClassInstanceModifierStrategyInterface>
+     * @return array<\SprykerSdk\Integrator\Builder\ClassModifier\ClassInstanceModifierStrategy\Wire\WireModifierStrategyInterface>
+     */
+    public function getWireClassInstanceModifierStrategies(): array
```
Comment (kalaputsik) [inline src/IntegratorFactory.php:724]: what about **getWireModifierStrategies** ?

File: src/IntegratorFactory.php [hunk 2]
```diff
@@ -672,57 +719,176 @@ public function getManifestExecutorStrategies(): array
...
+            $this->createWireClassInstanceReturnCollectionModifierStrategy(),
+            $this->createWireClassInstanceReturnExtendContainerModifierStrategy(),
+        ];
+    }
+
+    /**
+     * @return array<\SprykerSdk\Integrator\Builder\ClassModifier\ClassInstanceModifierStrategy\Unwire\UnwireModifierStrategyInterface>
      */
-    public function getClassInstanceModifierStrategies(): array
+    public function getUnwireClassInstanceModifierStrategies(): array
```
Comment (kalaputsik) [inline src/IntegratorFactory.php:738]: getUnwireModifierStrategies ?

**Review by kalaputsik — CHANGES_REQUESTED** [review]

**Review by kalaputsik — COMMENTED** [review]
getManifestExecutorStrategies -> getManifestStrategies ?

File: src/Builder/ClassModifier/ClassConstant/Unwire/UnwireModifierInterface.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+interface UnwireModifierInterface
+{
+    /**
+     * @param \SprykerSdk\Integrator\Transfer\ClassInformationTransfer $classInformationTransfer
+     * @param string $classNameToRemove
+     * @param string $targetMethodName
+     *
+     * @return \SprykerSdk\Integrator\Transfer\ClassInformationTransfer|null
+     */
+    public function unwireClassConstant(
```
Comment (kalaputsik) [inline src/Builder/ClassModifier/ClassConstant/Unwire/UnwireModifierInterface.php:23]: unwire ?

**Comment by kalaputsik** [comment]
**ClassConstant**
src/Builder/ClassModifier/ClassConstant/Unwire/UnwireModifier.php
src/Builder/ClassModifier/ClassConstant/Unwire/UnwireModifierInterface.php
execution in ClassBuilderFacade
**unwireClassConstant -> unwire**

src/Builder/ClassModifier/ClassConstant/Wire/WireModifier.php
src/Builder/ClassModifier/ClassConstant/Wire/WireModifierInterface.php
execution in ClassBuilderFacade
**wireClassConstant -> wire**

---

**ClassInstance**
src/Builder/ClassModifier/ClassInstance/Unwire/UnwireModifier.php
src/Builder/ClassModifier/ClassInstance/Unwire/UnwireModifierInterface.php
execution in ClassBuilderFacade
**unwireClassInstance -> unwire**

src/Builder/ClassModifier/ClassInstance/Wire/WireModifier.php
src/Builder/ClassModifier/ClassInstance/Wire/WireModifierInterface.php
execution in ClassBuilderFacade
**wireClassInstance -> wire**

---

**GlueRelationship**
src/Builder/ClassModifier/GlueRelationship/Unwire/UnwireModifier.php
src/Builder/ClassModifier/GlueRelationship/Unwire/UnwireModifierInterface.php
execution in ClassBuilderFacade
**unwireGlueRelationship -> unwire**

src/Builder/ClassModifier/GlueRelationship/Wire/WireModifier.php
src/Builder/ClassModifier/GlueRelationship/Wire/WireModifierInterface.php
execution in ClassBuilderFacade
**wireGlueRelationship -> wire**

File: src/IntegratorFactory.php (hunk 1 above)
Comment (VladislavStrelchenko) [reply src/IntegratorFactory.php:724]: Thanks ++

File: src/IntegratorFactory.php (hunk 2 above)
Comment (VladislavStrelchenko) [reply src/IntegratorFactory.php:738]: ++

**Review by kalaputsik — APPROVED** [review]

**Pull request merged**